oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/ClaimAccessor.java

```diff
@@ -98,8 +99,18 @@ default Instant getClaimAsInstant(String claim) {
 		if (Instant.class.isAssignableFrom(claimValue.getClass())) {
 			return (Instant) claimValue;
 		}
+		if (String.class.isAssignableFrom(claimValue.getClass())) {
+			try {
+				return Instant.parse((String) claimValue);
```

Comment:
Instead of `Instant.parse()` let's attempt both of these:

```
ZonedDateTime.parse(claimValue, DateTimeFormatter.ISO_INSTANT)
ZonedDateTime.parse(claimValue, DateTimeFormatter.ISO_ZONED_DATE_TIME)
```

If both of these do not parse along with the previous (existing) parse attempts than we'll keep with throwing `IllegalArgumentException` as is the current behaviour. I know I said I agreed to not throw for getters, but this is a special case IMO. If the claim exists, which is the case if the parse attempt proceed than we cannot just return `null` as we are basically silently ignoring the error and the _existing_ claim. This will just cause bigger issues for the user as tracking the root issue will make things more difficult.

Reply:
ZonedDateTime.parse(claimValue, DateTimeFormatter.ISO_INSTANT) doesn't work, but  Instant.from(DateTimeFormatter.ISO_INSTANT.parse((String)claimValue)) does.

The second one: DateTimeFormatter.ISO_ZONED_DATE_TIME works for UTC timestamps too (i.e. ending with Z), so I'm not sure if doing the ISO_INSTANT first is necessary.